pkg/backup/backup_pv_action.go

```diff
@@ -61,5 +62,19 @@ func (a *PVCAction) Execute(item runtime.Unstructured, backup *v1.Backup) (runti
 		GroupResource: kuberesource.PersistentVolumes,
 		Name:          pvc.Spec.VolumeName,
 	}
-	return item, []velero.ResourceIdentifier{pv}, nil
+	// remove dataSource if exists from prior restored CSI volumes
+	if pvc.Spec.DataSource != nil {
```

Comment:
I suggest we move the same to restore, specifically:
During restore, we judge if the DataSource/DataSourceRef is required (for CSI restore, it is required, for others it is not), if it is required, leave the fields there, otherwise, remove it as what we are doing here

Reasons:
1. In backup, we may have multiple BIAs, a BIA doesn't know what others is doing or will do, so if not necessary, a BIA should not change the data comparing to the source, otherwise, we don't know what other BIAs will react
2. As the principle, if not necessary, the backup procedure should not change anything comparing to the source. This principle guarantees the consistency and coherence of the restore data and finally guarantees the validation of the restore
3. The current problem is right a requirement that we need to take care of the  DataSource/DataSourceRef, just like what we do for Status fields. Therefore, it looks more appropriate to put it into restore

On the other hand, I am not seeing any direct error the current change will cause, so I will leave this to more members for judgements, @reasonerjt @blackpiglet @shubham-pampattiwar

Reply:
Keeping the resources unchanged in backup is reasonable, but I also couldn't spot any scenarios broken by the current change.

Reply:
@Lyndon-Li the problem with clearing it on restore is that restore is where some plugins might add it. For example, CSI restore plugin adds it, so if the built-in RIA clears it, we could have a race condition, and there's no way of knowing within one plugin action whether another action has been called and/or will be called. Are there any use cases where having this in backup is needed? This field is only used for initial content creation for a new PVC. This source provided the content for the PVC at creation time. It will not have any content that was added/modified after creation -- for that we need the *current* content, via kopia/restic, CSI snapshot, or native snapshot. So I'm not sure we ever need/want this in the backup.

Reply:
@sseago Yes, indeed, if we want to put the clear code in restore, we need to make sure that the fields are not added by a CSI RIA plugin.  One idea is we put the clear code at early stage of `restoreItem`, e.g., somewhere[ around here](https://github.com/vmware-tanzu/velero/blob/d298c6d0a2284d0e0444998efd3ad94472a312c2/pkg/restore/restore.go#L1184), none of the RIAs has been called at this place, so it is safe to remove the fields without a check.

> Are there any use cases where having this in backup is needed?

Personally, I don't see any need for them during in backup at present, so technically, I don't see any error will happen if we keep the current clear code in backup.
Merely, I just think it is more appropriate to have the code in restore (as the reasons mentioned above). Therefore, I raise this as a suggestion, I am fine with either way as the final decision.

Reply:
> as for most use cases, restore is more appropriate, but in cases where a field will never be useful for restore (or for RIA funcs), then removing on backup is fine

OK, this sounds reasonable